crawler/src/main/scala/io/scalac/newspaper/crawler/failure/FailureHandler.scala

```diff
@@ -1,17 +1,71 @@
 package io.scalac.newspaper.crawler.failure
 
-import akka.Done
-import akka.stream.scaladsl.Sink
-import com.typesafe.scalalogging.StrictLogging
-import io.scalac.newspaper.crawler.fetching.FetchingFlow.{URLFetchingException, URLFetchingResult, URLFetchingTimeout, URLNotFetched}
+import akka.actor.{ActorLogging, Props}
+import akka.persistence.{PersistentActor, SnapshotOffer}
+import io.scalac.newspaper.crawler.failure.FailureHandler._
+import io.scalac.newspaper.crawler.fetching.FetchingFlow.{URLFetched, URLFetchingResult}
+import io.scalac.newspaper.crawler.urls.URLsStore
 
-import scala.concurrent.Future
+import scala.collection.mutable
 
-trait FailureHandler extends StrictLogging {
+/**
+  * Created by rsekulski on 30.06.2017.
+  */
+object FailureHandler {
+  case class FailureReported(url: String, result: URLFetchingResult)
 
-  def handleFailure: Sink[URLFetchingResult, Future[Done]] = Sink.foreach {
-    case URLNotFetched(url, responseCode, content) => logger.error(s"Unable to fetch: $url, response code: $responseCode, response body: $content")
-    case URLFetchingTimeout(url) => logger.error(s"Timeout when trying to fetch: $url")
-    case URLFetchingException(url, ex) => logger.error(s"Exception when trying to fetch url: $url", ex)
+  def props(urlsStore: URLsStore, config: FailureHandlerConfig = defaultConfig) = Props(new FailureHandler(urlsStore, config))
+
+  case class FailureHandlerConfig(failuresLimit: Int, snapshotInterval: Int)
+  val defaultConfig =  FailureHandlerConfig(3, 50)
+
+  type FailuresStore = mutable.Map[String, Seq[URLFetchingResult]]
+
+  case class GetFailures(url: String)
+  case class URLFailures(failures: Option[Seq[URLFetchingResult]])
```

Comment:
You'll always get either Some(non-empty Seq) or None. You asked, because you would like to propose other model for this message?